File: html/updateprofile.js
"use strict";

function submit(){
    let file = document.getElementById("ppic").files[0];
    if(!file){
        file = new Blob();
    }
    let R = new FileReader();
    let path = document.location.pathname;
    R.addEventListener("load", () => {
        let profilepic = btoa(R.result);    //do base64 encoding
        let fname = document.getElementById("fname").value;
        let lname = document.getElementById("lname").value;
        let dob = document.getElementById("birthdate").value;
        let J = {
            firstName: fname,
            lastName: lname,
            birthDate: dob,
            pic: profilepic
        };
        fetch( path,
            {   method: "POST",
                body: JSON.stringify(J)
            }
        ).then( (resp) => {
            //can also use text(), blob(), or arrayBuffer()
            resp.json().then( (J) => {
                console.log("Server said:",J);
            });
        }).catch( (err) => {
            console.log("Uh oh",err);
        })
    });
    R.readAsBinaryString(file);
}

File: src/updateprofile.py
import tornado.web
import json
import base64
import Profile

class Handler(tornado.web.RequestHandler):
    def get(self):
        p = self.request.path.split("/")
        uname = p[2]
        info = Profile.D[uname]
        self.render("updateprofile.html", fname = info["fname"], lname = info["lname"],
                        birthDate = info["birthDate"], ppic = info["ppic"])
    
    def post(self):
        J=json.loads(self.request.body)
        firstName = J["firstName"]
        lastName = J["lastName"]
        dob = J["birthDate"]
        p = self.request.path.split("/")
        uname = p[2]
        info = Profile.D[uname]
        info["fname"] = firstName
        info["lname"] = lastName
        info["birthDate"] = dob
        if (not J["pic"]):
            pass
        else:
            ppic = J["pic"]
            info["ppic"] = "data:image/png;base64," + ppic
        print("WE GOT:",firstName,lastName,dob,ppic[:20])
        resp={"ok": True}
        self.write( json.dumps(resp) )
    
